feat(ftc): add disclaimer to cc form

brendanhsentry opened the pull request:
closes https://linear.app/getsentry/issue/BIL-501/add-disclaimer-to-payment-method-step-in-checkout and https://linear.app/getsentry/issue/BIL-502/add-disclaimer-to-billing-details-modal

<img width="613" alt="Screenshot 2025-04-23 at 1 29 37 PM" src="https://github.com/user-attachments/assets/12194290-d32b-4e12-95c7-77dd92d3a1f4" />
 <img width="679" alt="Screenshot 2025-04-23 at 1 29 55 PM" src="https://github.com/user-attachments/assets/6672a405-e696-4f5a-8484-132f84c3db55" />

brendanhsentry committed:
feat(ftc): add disclaimer to cc form

brendanhsentry committed:
Merge branch 'master' into brendan/ftc-cc-disclaimer

isabellaenriquez commented on static/gsApp/components/creditCardForm.tsx:
```diff
@@ -231,6 +237,19 @@ function CreditCardForm({
               stripe: <ExternalLink href="https://stripe.com/" />,
             })}
           </small>
+          {location !== null && location !== undefined && (
```
```suggestion
          {location && (
```

brendanhsentry replied:
`location` is 0 in the checkout page so it would hide the disclaimer

isabellaenriquez replied:
oh right lol

isabellaenriquez commented on static/gsApp/components/creditCardForm.tsx:
```diff
@@ -231,6 +237,19 @@ function CreditCardForm({
+          {location !== null && location !== undefined && (
+            <FinePrint>
+              {tct(
+                'By clicking [buttonText], you authorize Sentry to automatically charge you recurring subscription fees and applicable pay-as-you-go fees. Recurring charges occur at the start of your selected billing cycle for subscription fees and monthly for pay-as-you-go fees. You may cancel your subscription at any time [here:here].',
```
do we need to say "on-demand" for pre-am3 customers?

brendanhsentry replied:
yeah, good point

brendanhsentry committed:
feat(ftc): update api call to include location (#90246)

closes
https://linear.app/getsentry/issue/BIL-507/add-parameter-to-api-for-specifying-location-where-payment-method-was

depends on https://github.com/getsentry/getsentry/pull/17280

adds `ftcConsentLocation` to `CustomerDetailsEndpoint` call

note: I plan to merge this PR into
https://github.com/getsentry/sentry/pull/90241 to roll out the frontend
changes at the same time.

brendanhsentry committed:
add budget term to disclaimer
